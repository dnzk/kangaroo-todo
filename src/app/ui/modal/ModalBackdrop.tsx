import style from "./modal.module.scss"
import Modal from "./Modal"

type BackdropProps = {
  children?: React.JSX.Element,
}

export default function ModalBackdrop({ children }: BackdropProps) {
  return (
    <>
      <div className={style.backdrop} />
      <Modal>
        {children}
      </Modal>
    </>
  )
}
