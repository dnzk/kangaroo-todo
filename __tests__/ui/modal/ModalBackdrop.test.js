import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'
import ModalBackdrop from "../../../src/app/ui/modal/ModalBackdrop"

describe('ModalBackdrop', () => {
  it('renders correctly', () => {
    const { container } = render(<ModalBackdrop />)
    expect(container).toMatchSnapshot()
  })

  it('renders children when supplied', () => {
    render(<ModalBackdrop><p>Modal goes here</p></ModalBackdrop>)
    const input = screen.getByText('Modal goes here')
    expect(input).toBeInTheDocument()
  })
})
