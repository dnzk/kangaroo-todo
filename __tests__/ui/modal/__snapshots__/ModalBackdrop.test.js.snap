// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalBackdrop renders correctly 1`] = `
<div>
  <div
    class="backdrop"
  />
  <div
    class="modal"
  />
</div>
`;
